### PR TITLE
PLAT-94881: Add JSDoc to Agate

### PR DESCRIPTION
**Pull request by webOS101**
This PR adds or corrects the inline documentation.  There are some placeholders for text I wasn't sure about.  Also, there is a lot of inconsistency about what is exported from various components.  We may want to address this in a future PR.  I did not, in general, add jsdoc for props where they were missing.  This, too, will have to be addressed in a future PR.

### CODE REVIEW

File: SliderButton/SliderButton.js [hunk 1]
```diff
@@ -105,17 +126,33 @@ const SliderButtonBase = kind({
 	}
 });
 
-
+/**
+ * Agate specific item behaviors to apply to [SliderButton]{@link agate/SliderButton.SliderButtonBase}.
```
Comment (sugardave) [inline SliderButton/SliderButton.js]: Can we nix "item" here?

File: Switch/Switch.js [hunk 2]
```diff
@@ -25,16 +25,16 @@ import componentCss from './Switch.module.less';
...
- * @extends agate/ToggleIcon.ToggleIcon
+ * @extends ui/ToggleIcon.ToggleIconBase
  * @ui
  * @public
  */
 const SwitchBase = kind({
 	name: 'Switch',
 
-	propTypes: /** @lends agate/Switch.Switch.prototype */ {
+	propTypes: /** @lends agate/Switch.SwitchBase.prototype */ {
```
Comment (sugardave) [inline Switch/Switch.js]: In some of the other changes, you went from `agate/Component.ComponentBase.prototype` to `agate/Component.Component.prototype`, is there a reason for them to differ?

File: Switch/Switch.js [hunk 3]
```diff
@@ -71,26 +71,44 @@ const SwitchBase = kind({
...
-			>
-				{children}
-			</ToggleIconBase>
+			/>
 		);
 	}
 });
 
+/**
+ * Agate specific item behaviors to apply to [Switch]{@link agate/Switch.SwitchBase}.
```
Comment (sugardave) [inline Switch/Switch.js]: Can we nix "item" here, too?

File: ThumbnailItem/ThumbnailItem.js [hunk 4]
```diff
@@ -6,10 +17,19 @@ import {SlotItem} from '@enact/ui/SlotItem';
...
+ * @memberof agate/ThumbnailItem
+ * @extends ui/SlotItem.SlotItem
+ * @ui
+ * @public
+ */
 const ThumbnailItemBase = kind({
 	name: 'ThumbnailItem',
 
-	propTypes: {
+	propTypes: /** @lends agate/ThumbnailItem.ThumbnailItemBase.prototype */ {
```
Comment (sugardave) [inline ThumbnailItem/ThumbnailItem.js]: Same ? about which prototype to lend

File: ToggleButton/ToggleButton.js [hunk 5]
```diff
@@ -33,7 +33,7 @@ import css from './ToggleButton.module.less';
 const ToggleButtonBase = kind({
 	name: 'ToggleButton',
 
-	propTypes: /** @lends agate/Button.ButtonBase.prototype */ {
+	propTypes: /** @lends agate/ToggleButton.ToggleButtonBase.prototype */ {
```
Comment (sugardave) [inline ToggleButton/ToggleButton.js]: Here, too (lends which prototype?)

File: FullscreenPopup/FullscreenPopup.js [hunk 6]
```diff
@@ -9,9 +17,17 @@ import Skinnable from '../Skinnable';
...
+ *
+ * @class FullscreenPopupBase
+ * @memberof agate/FullscreenPopup
+ * @ui
+ * @public
+ */
 const FullscreenPopupBase = kind({
 	name: 'FullscreenPopup',
-	propTypes: {
+	propTypes: /** @lends agate/FullscreenPopup.FullscreenPopupBase.prototype */ {
```
Comment (sugardave) [inline FullscreenPopup/FullscreenPopup.js]: One more.

File: AgateDecorator/AgateDecorator.js [hunk 7]
```diff
@@ -24,31 +25,125 @@ import css from './AgateDecorator.module.less';
...
+	 * @type {Boolean}
+	 * @default true
+	 * @public
+	 */
+	customSkin: true,
+
+	/**
+	 * Enables a floating layer for popup components.
+	 *
+	 * If not applied, app will be responsible for applying the decorator.
```
[the remark this answers is not in the document]
Comment (webOS101) [reply AgateDecorator/AgateDecorator.js]: Sure?

File: SliderButton/SliderButton.js (hunk 1 above)
Comment (webOS101) [reply SliderButton/SliderButton.js]: Good plan.

File: Switch/Switch.js (hunk 2 above)
Comment (webOS101) [reply Switch/Switch.js]: Yes, in those cases there is no (documented) `ComponentBase`.

File: Switch/Switch.js (hunk 3 above)
Comment (webOS101) [reply Switch/Switch.js]: :+1:

File: ThumbnailItem/ThumbnailItem.js (hunk 4 above)
Comment (webOS101) [reply ThumbnailItem/ThumbnailItem.js]: This seems right to me.

File: ToggleButton/ToggleButton.js (hunk 5 above)
Comment (webOS101) [reply ToggleButton/ToggleButton.js]: This is right, too.

File: FullscreenPopup/FullscreenPopup.js (hunk 6 above)
Comment (webOS101) [reply FullscreenPopup/FullscreenPopup.js]: Also right.

File: Switch/Switch.js (hunk 2 above)
Comment (sugardave) [reply Switch/Switch.js]: Ah, got it.

**Comment by webOS101** [comment]
For ease of testing, you can link agate into the enact/packages directory for parsing.

**Pull request merged**